api/v2/api_test.go

```diff
@@ -240,11 +240,13 @@ func TestPostSilencesHandler(t *testing.T) {
 			expectedCode int
 		}{
 			{
-				"with an non-existent silence ID - it returns 404",
+				// This can happen when updating an expired silence that has been GC'd.
```

Comment:
These are new semantics for silences, so please review with care. If considered problematic then I will reconsider how `silences.Set` works for missing IDs.